grafana/resource_organization.go

```diff
@@ -185,53 +185,67 @@ func ReadUsers(d *schema.ResourceData, meta interface{}) error {
 }
 
 func UpdateUsers(d *schema.ResourceData, meta interface{}) error {
-	oldUsers, newUsers := collectUsers(d)
-	changes := changes(oldUsers, newUsers)
+	stateUsers, configUsers, err := collectUsers(d)
+	if err != nil {
+		return err
+	}
+	changes := changes(stateUsers, configUsers)
 	orgId, _ := strconv.ParseInt(d.Id(), 10, 64)
-	changes, err := addIds(d, meta, changes)
+	changes, err = addIdsToChanges(d, meta, changes)
 	if err != nil {
 		return err
 	}
 	return applyChanges(meta, orgId, changes)
 }
 
-func collectUsers(d *schema.ResourceData) (map[string]OrgUser, map[string]OrgUser) {
+func collectUsers(d *schema.ResourceData) (map[string]OrgUser, map[string]OrgUser, error) {
 	roles := []string{"admins", "editors", "viewers"}
-	oldUsers, newUsers := make(map[string]OrgUser), make(map[string]OrgUser)
+	stateUsers, configUsers := make(map[string]OrgUser), make(map[string]OrgUser)
 	for _, role := range roles {
 		roleName := strings.Title(role[:len(role)-1])
-		old, new := d.GetChange(role)
-		for _, u := range old.([]interface{}) {
-			oldUsers[u.(string)] = OrgUser{0, u.(string), roleName}
+		// Get the lists of users read in from Grafana state (old) and configured (new)
+		state, config := d.GetChange(role)
+		for _, u := range state.([]interface{}) {
+			stateUsers[u.(string)] = OrgUser{0, u.(string), roleName}
 		}
-		for _, u := range new.([]interface{}) {
-			newUsers[u.(string)] = OrgUser{0, u.(string), roleName}
+		for _, u := range config.([]interface{}) {
+			email := u.(string)
```

Comment:
This is helpful, I think you should do it for above as well.

Reply:
Good point- best to play it safe